### ISSUE TITLE
ESBuild cannot resolve tsconfig extends that use node16 "exports"

### ISSUE THREAD

**Issue by rossipedia** [evanw/esbuild#3437]
Imagine you have a shared set of Typescript configurations that you use for different project types, laid out something like:

```
.
├── configs
│   ├── tsconfig.environment-agnostic.json
│   ├── tsconfig.node-cjs.json
│   ├── tsconfig.node-esm.json
│   ├── tsconfig.react-published.json
│   └── tsconfig.react.json
├── internal
│   └── tsconfig.base.json
└── package.json
```

And in that `package.json` you're utilizing explicit [package entry points](https://nodejs.org/dist/latest-v20.x/docs/api/packages.html#package-entry-points) to make those available using a friendlier name:

```json
  "name": "@scope/tsconfigs",
  "exports": {
    "./package.json": "./package.json",
    "./*": "./configs/tsconfig.*.json"
  },
```

This allows you to write a `tsconfig.json` like so:

```json
{
  "extends": "@scope/tsconfigs/react",
  "compilerOptions": {
    ...
  }
}
```

With the intention of extending the config defined in `configs/tsconfig.react.json` in the `tsconfigs` package.

Typescript itself will correctly resolve the config file in extends, whereas `esbuild` issues the following warning:

```
▲ [WARNING] Cannot find base config file "@scope/tsconfigs/react" [tsconfig.json]

    tsconfig.json:2:13:
```

**Comment by mebest100** [comment]
Do not use esbuild anymore. The author is a rubbish and shit!    Bad and unclear docs and huge bugs and no reply even then, go to hell, shit author!

**Comment by evanw** [comment]
Sorry about that random comment (which was deleted). Anyone can post anything on GitHub.

Anyway, thanks for the report. I took a look and I was not able to reproduce this given the information you provided. [Here's what I tried](https://esbuild.github.io/try/#YgAwLjE5LjQALS1idW5kbGUAZQBleGFtcGxlLnRzeAB0ZXN0KDxkaXYvPikAAHRzY29uZmlnLmpzb24AewogICJleHRlbmRzIjogIkBzY29wZS90c2NvbmZpZ3MvcmVhY3QiLAogICJjb21waWxlck9wdGlvbnMiOiB7CiAgfQp9AABub2RlX21vZHVsZXMvQHNjb3BlL3RzY29uZmlncy9wYWNrYWdlLmpzb24AewogICJuYW1lIjogIkBzY29wZS90c2NvbmZpZ3MiLAogICJleHBvcnRzIjogewogICAgIi4vcGFja2FnZS5qc29uIjogIi4vcGFja2FnZS5qc29uIiwKICAgICIuLyoiOiAiLi9jb25maWdzL3RzY29uZmlnLiouanNvbiIKICB9Cn0AAG5vZGVfbW9kdWxlcy9Ac2NvcGUvdHNjb25maWdzL2NvbmZpZ3MvdHNjb25maWcucmVhY3QuanNvbgB7CiAgImNvbXBpbGVyT3B0aW9ucyI6IHsKICAgICJqc3hGYWN0b3J5IjogIndvcmtzIgogIH0KfQ) which seems to work fine. What additional information is necessary to reproduce your problem?

Note that an issue like this has already been fixed last April, in esbuild version 0.17.17: #3058. You will see this behavior if you're using an old version of esbuild. In that case you should update esbuild to get the fix.

**Comment by evanw** [comment]
I'm closing this issue due to lack of a follow-up reply. We can reopen it in the future if more information is provided.
